nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -4250,4 +4284,92 @@ mod tests {
              found these zones not in service: {not_in_service:?}"
         );
     }
+
+    // Verify that every blueprint-related table contains ≥1 row for `blueprint_id`.
+    // Complements `ensure_blueprint_fully_deleted`.
+    async fn ensure_blueprint_fully_populated(
+        datastore: &DataStore,
+        blueprint_id: BlueprintUuid,
+    ) {
+        let conn = datastore.pool_connection_for_tests().await.unwrap();
+
+        macro_rules! query_count {
+            ($table:ident, $blueprint_id_col:ident) => {{
+                use nexus_db_schema::schema::$table::dsl;
+                let result = dsl::$table
+                    .filter(
+                        dsl::$blueprint_id_col
+                            .eq(to_db_typed_uuid(blueprint_id)),
+                    )
+                    .count()
+                    .get_result_async(&*conn)
+                    .await;
+                (stringify!($table), result)
+            }};
+        }
+
+        // Tables prefixed with `bp_` that are *not* specific to a single blueprint
+        // and therefore intentionally ignored.
+        let tables_ignored: BTreeSet<_> = ["bp_target"].into_iter().collect();
+
+        let mut tables_checked = BTreeSet::new();
+        for (table_name, result) in [
+            query_count!(blueprint, id),
+            query_count!(bp_sled_metadata, blueprint_id),
+            query_count!(bp_omicron_dataset, blueprint_id),
+            query_count!(bp_omicron_physical_disk, blueprint_id),
+            query_count!(bp_omicron_zone, blueprint_id),
+            query_count!(bp_omicron_zone_nic, blueprint_id),
+            query_count!(bp_clickhouse_cluster_config, blueprint_id),
+            query_count!(bp_clickhouse_keeper_zone_id_to_node_id, blueprint_id),
+            query_count!(bp_clickhouse_server_zone_id_to_node_id, blueprint_id),
+            query_count!(bp_oximeter_read_policy, blueprint_id),
+            query_count!(bp_pending_mgs_update_sp, blueprint_id),
+            query_count!(bp_pending_mgs_update_rot, blueprint_id),
+            query_count!(bp_pending_mgs_update_rot_bootloader, blueprint_id),
+        ] {
+            let count: i64 = result.unwrap();
+
+            // Pending MGS update tables are only populated when the blueprint includes firmware updates.
+            //  The representative blueprint doesn't, so allow zero rows for these tables.
+            if matches!(
+                table_name,
+                "bp_pending_mgs_update_sp"
+                    | "bp_pending_mgs_update_rot"
+                    | "bp_pending_mgs_update_rot_bootloader"
+            ) {
+                tables_checked.insert(table_name);
+                continue;
+            }
```

Comment:
I thought the representative collection did have some kind of update.  If not, shouldn't it?  Maybe it could have one of each type (probably with different baseboard ids)?  Our planner would never do that but I don't believe it's actually invalid.